libraries/AP_HAL/utility/RingBuffer.h

```diff
@@ -72,7 +72,31 @@ class ByteBuffer {
       read len bytes without advancing the read pointer
     */
     uint32_t peekbytes(uint8_t *data, uint32_t len);
-    
+
+    // Similar to peekbytes(), but will fill out IoVec struct with
+    // both parts of the ring buffer if wraparound is happpening, or
+    // just one part. Returns the number of parts written to.
+    struct IoVec {
+        uint8_t *data;
+        uint32_t len;
+    };
+    int peekiovec(IoVec vec[2], uint32_t len);
```

Comment:
> return type of the other method? I didn't get that. It can just be unsigned int, dont need to force size 32 and if it is to force a size, 8 looks like a better choice.

it's ok to use uint8_t there